eras/conway/impl/test/Test/Cardano/Ledger/Conway/GovSpec.hs

```diff
@@ -0,0 +1,128 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Cardano.Ledger.Conway.GovSpec where
+
+import Cardano.Ledger.BaseTypes
+import Cardano.Ledger.Conway
+import Cardano.Ledger.Conway.Governance
+import Cardano.Ledger.Shelley.LedgerState
+import Data.Default.Class (Default (..))
+import Data.Maybe
+import Lens.Micro
+import Test.Cardano.Ledger.Common
+import Test.Cardano.Ledger.Conway.ImpTest
+
+spec :: Spec
+spec =
+  describe "Proposals always have valid previous actions" $ do
+    context "Invalid proposals are rejected" $ do
+      itM @Conway "Lookup snapshots" $ do
+        constitutionHash <- freshSafeHash
+        let constitutionAction =
+              NewConstitution
+                SNothing
+                ( Constitution
+                    ( Anchor
+                        (fromJust $ textToUrl "constitution.0")
+                        constitutionHash
+                    )
+                    SNothing
+                )
+        gaidConstitutionProp <- submitProposal constitutionAction
+        let constitutionActionNext =
+              NewConstitution
+                (SJust $ PrevGovActionId (gaidConstitutionProp {gaidGovActionIx = GovActionIx 1})) -- make it wrong
+                ( Constitution
+                    ( Anchor
+                        (fromJust $ textToUrl "constitution.1")
+                        constitutionHash
+                    )
+                    SNothing
+                )
+        submitFailingProposal constitutionActionNext
+      itM @Conway "Lookup snapshots" $ do
```

Comment:
This test wasn't particularly useful, since it was failing for the same reason as the test above, namely missing govActionId. we could instead make it more interesting and expect it to fail for the reason of mismatched purpose